.circleci/config.yml

```diff
@@ -26,107 +34,197 @@ executors:
+      - test-assembly-unix
+      - deploy-snapshot-unix
 
-  test-assembly-linux-arm64-zip:
-    executor: linux-arm64
-    working_directory: ~/typedb
+  test-deploy-snapshot-mac-arm64:
+    executor: mac-arm64
+    steps:
+      - checkout
+      - install-deps-brew:
+          bazel-arch: arm64
+      - test-assembly-unix
+      - deploy-snapshot-unix
+
+  # per platform release deployment
+  test-deploy-release-linux-x86_64:
+    executor: linux-x86_64-amazonlinux-2
     steps:
       - checkout
-      - install-bazel-linux-arm64
-      - run: bazel test //test/assembly:assembly --test_output=errors
+      - install-deps-yum:
+          bazel-arch: amd64
+      - test-assembly-unix
+      - deploy-release-unix
 
-  test-assembly-linux-x86_64-zip:
-    executor: linux-x86_64
-    working_directory: ~/typedb
+  deploy-docker:
+    executor: linux-x86_64-amazonlinux-2
     steps:
       - checkout
-      - install-bazel-linux-x86_64
-      - run: bazel test //test/assembly:assembly --test_output=errors
+      - install-deps-yum:
+          bazel-arch: amd64
+      - deploy-docker
 
-  test-assembly-windows-zip:
-    executor:
-      name: win/default
-      shell: cmd.exe
-    working_directory: ~/typedb
+
+  # generic
+  release-cleanup:
+    executor: linux-x86_64-ubuntu-2204
     steps:
       - checkout
-      - run: .circleci\windows\prepare.bat
-      - run: .circleci\windows\test-assembly.bat
+      - run: |
+          apt update -y
+          apt install -y git
+          git push --delete https://$REPO_GITHUB_TOKEN@github.com/typedb/typedb-driver.git $CIRCLE_BRANCH
 
 
 workflows:
-#  typedb:
-#    jobs:
-#      - test-assembly-mac-x86_64-zip:
-#          filters:
-#            branches:
-#              only:
-#                - master
-#                - development
-#      - test-assembly-mac-arm64-zip:
-#          filters:
-#            branches:
-#              only:
-#                - master
-#                - development
-#      - test-assembly-linux-x86_64-zip:
-#          filters:
-#            branches:
-#              only:
-#                - master
-#                - development
-#      - test-assembly-linux-arm64-zip:
-#          filters:
-#            branches:
-#              only:
-#                - master
-#                - development
-#      - test-assembly-windows-zip:
-#          filters:
-#            branches:
-#              only:
-#                - master
-#                - development
+  typedb-snapshot:
+    jobs:
+      - test-deploy-snapshot-linux-x86_64:
+          filters:
+            branches:
+              only: [ 3.0-dev ]
+
+      - test-deploy-snapshot-linux-arm64:
+          filters:
+            branches:
+              only: [ 3.0-dev ]
+
+      - test-deploy-snapshot-mac-x86_64:
+          filters:
+            branches:
+              only: [ 3.0-dev ]
+
+      - test-deploy-snapshot-mac-arm64:
+          filters:
+            branches:
+              only: [ 3.0-dev ]
```

Comment:
The circleci validator failed when I used `3.0` because it expected a string, received a double.